cookbooks/omnibus-supermarket/libraries/config.rb

```diff
@@ -1,58 +1,223 @@
-require 'fileutils'
-require 'securerandom'
+require 'mixlib/config'
+require 'chef/mixin/deep_merge'
 
-# Manages configuration
-class Supermarket
+module Supermarket
```

Comment:
> So, did we decide to change the config file format? 

No, there was no discussion around changing the configuration file format.  I chose to do it completely autonomously. 

> The reason it was done the way it was done in the first place is that if we do it like this, we have to define and document everything in one or two places: the attributes file and here.

I agree that we should probably only document what does what in one place, namely the attributes file. I'd be happy to write some words in the config module that explains where to go for the canonical source of truth.

>This also makes it so that if you add a configurable value in the app, you must add it here as well, making some more maintenance pain.

I suppose that's true, but it's also how every other omnibus package that we ship does things.  For people that are used to configuring our applications, having an alternative configuration syntax is strange and confusing.  I could disable strict mode in the config to allow new attributes to be used without explicitly defining them in the config but I haven't investigated the potential drawbacks of that approach.

> I'm cool with having the conversation around completely changing how we configure this, and I'm totally open to it, but it seems like it's out of scope for this PR. Is there anything in making a combined mode that requires us to change this as well?

I agree that it is probably outside of the scope for this PR, however, there is one dependency on it though. In combined mode it will create/re-use a database password, something that isn't currently supported.